objgraph.py

```diff
@@ -355,6 +356,91 @@ def show_growth(limit=10, peak_stats=None, shortnames=True, file=None,
             file.write('%-*s%9d %+9d\n' % (width, name, count, delta))
 
 
+def get_ids(skip_update=False, limit=10, sortby='deltas',
+            OLD_IDS=collections.defaultdict(set),
+            CURRENT_IDS=collections.defaultdict(set),
+            NEW_IDS=collections.defaultdict(set)):
+    """Show the increase in object counts since last call to this function
+    and returns the memory address ids for old, current, and new objects.
+
+    Returns: [OLD_IDS, CURRENT_IDS, NEW_IDS]
+
+    OLD_IDS: a dictionary which stores sets of ids under the keys of object
+    type. These are the objects that were stored the last time the function was
+    called.
+
+    CURRENT_IDS: a dictionary which stores sets of ids under the keys of object
+    type. These are the objects that are stored when the function is called
+    now.
+
+    NEW_IDS: a dictionary which stores sets of ids under the keys of object
+    type. These are the objects that were created between the last time this
+    function was called and when it was called now.
+
+    sortby: This is the column that you want to sort by in descending order.
+    Possible values are: 'old', 'current', 'new', 'deltas'
+
+    limit: the maximum number of rows that you want to print data for
+
+    if ``skip_update`` is True, the sets of [OLD_IDS, CURRENT_IDS, NEW_IDS]
+    will be returned from when the function was last run without examining the
+    objects currently iin memory.
+
+    The caveats documented in :func:`growth` apply.
+
+    Example:
+
+        >>> get_ids()
+        ======================================================================
+        Type                    Old_ids  Current_ids      New_ids Count_Deltas
+        ======================================================================
+        weakref                    3807         3864          +57          +57
+        dict                       6892         6947          +73          +55
+        frame                        34           70          +53          +36
+        ...
+
+    """
+    if skip_update:
+        return [OLD_IDS, CURRENT_IDS, NEW_IDS]
+    gc.collect()
+    objects = gc.get_objects()
+    for k in OLD_IDS.keys():
+        OLD_IDS[k].clear()
+    for k, v in CURRENT_IDS.items():
+        OLD_IDS[k].update(v)
+    for k in CURRENT_IDS.keys():
+        CURRENT_IDS[k].clear()
+    for o in objects:
+        CURRENT_IDS[type(o).__name__].add(id(o))
+    for k in NEW_IDS.keys():
+        NEW_IDS[k].clear()
+    rows = []
+    for class_name in CURRENT_IDS.keys():
+        new_ids_set = CURRENT_IDS[class_name] - OLD_IDS[class_name]
+        NEW_IDS[class_name].update(new_ids_set)
```

Comment:
You're clearing NEW_IDs before this loop, and class names will not repeat, so you can do `NEW_IDS[...] = new_ids_set` and skip the update.